ngi_reports/reports/element_project_summary.py

```diff
@@ -0,0 +1,205 @@
+"""Module for producing the Element Project Summary Report"""
+
+import os
+import ngi_reports.reports.project_summary
+
+from collections import OrderedDict
+from string import ascii_uppercase as alphabets
+
+
+class Report(ngi_reports.reports.project_summary.Report):
+    def __init__(self, LOG, working_dir, **kwargs):
+        super(Report, self).__init__(LOG, working_dir, **kwargs)
+
+    def generate_report_template(self, proj, template, support_email):
+        # Check and exit if signature not provided
+        if not self.signature:
+            self.LOG.error(
+                "It is required to provide Signature/Name while generating 'project_summary' report, see -s opition in help"
+            )
+            raise SystemExit
+        else:
+            self.report_info["signature"] = self.signature
```

Comment:
Could be worth moving this to a function in the parent class.

Reply:
Good point!